dbms/src/Interpreters/Join.cpp

```diff
@@ -1190,27 +1230,39 @@ void NO_INLINE joinBlockImplTypeCase(
             if (it != internal_map.end())
             {
                 it->getMapped().setUsed();
-                Adder<KIND, STRICTNESS, Map>::addFound(
+                block_full = Adder<KIND, STRICTNESS, Map>::addFound(
                     it,
                     num_columns_to_add,
                     added_columns,
                     i,
                     filter.get(),
                     current_offset,
                     offsets_to_replicate.get(),
-                    right_indexes);
+                    right_indexes,
+                    probe_process_info);
             }
             else
-                Adder<KIND, STRICTNESS, Map>::addNotFound(
+                block_full = Adder<KIND, STRICTNESS, Map>::addNotFound(
                     num_columns_to_add,
                     added_columns,
                     i,
                     filter.get(),
                     current_offset,
-                    offsets_to_replicate.get());
+                    offsets_to_replicate.get(),
+                    probe_process_info);
             keyHolderDiscardKey(key_holder);
         }
+
+        // if block_full is true means that the current offset is greater than max_block_size, we need break the loop.
+        if (block_full)
+        {
+            break;
+        }
```

Comment:
it is useless now?

Reply:
> it is useless now?

It is necessary because we have to break immediately to make sure that end_row is updated to the correct value.